apps/server/src/modules/lti-tool/service/lti-tool.service.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { Test, TestingModule } from '@nestjs/testing';
+import { LtiToolDO } from '@shared/domain';
+import { LtiToolRepo } from '@shared/repo';
+import { ltiToolDOFactory } from '@shared/testing';
+import { LtiToolService } from './lti-tool.service';
+
+describe('LtiToolService', () => {
+	let module: TestingModule;
+	let service: LtiToolService;
+
+	let ltiToolRepo: DeepMocked<LtiToolRepo>;
+
+	beforeAll(async () => {
+		module = await Test.createTestingModule({
+			providers: [
+				LtiToolService,
+				{
+					provide: LtiToolRepo,
+					useValue: createMock<LtiToolRepo>(),
+				},
+			],
+		}).compile();
+
+		service = module.get(LtiToolService);
+		ltiToolRepo = module.get(LtiToolRepo);
+	});
+
+	afterAll(async () => {
+		await module.close();
+	});
+
+	describe('findByClientIdAndIsLocal', () => {
+		describe('when searching for an lti tool', () => {
+			const setup = () => {
+				const ltiTool: LtiToolDO = ltiToolDOFactory.buildWithId();
+
+				ltiToolRepo.findByClientIdAndIsLocal.mockResolvedValue(ltiTool);
```

Comment:
If possible please always use mockResolvedValueOnce instead of mockResolvedValue. It is use one more files in this PR.